pkg/ipamd/ipamd.go

```diff
@@ -655,7 +658,21 @@ func (c *IPAMContext) StartNodeIPPoolManager() {
 }
 
 func (c *IPAMContext) updateIPPoolIfRequired(ctx context.Context) {
-	c.askForTrunkENIIfNeeded(ctx)
+	if c.enablePodENI {
+		if labels, err := c.getNodeLabels(ctx); err == nil {
```

Comment:
`getNodeLabels()` will be called every 30 seconds to ensure the correct node labels are added to nodes. This shouldn't be expensive considering it calls to cached client. We call this method only once for two labeling checks (trunk and eniConfig). One note to this change can be that we should revisit this change if we are seeing any performance impact to API server in large clusters due to this new call from every aws-node replica.